pipeline_dsl/resources/pool.py

```diff
@@ -1,12 +1,16 @@
-class Pool:
-    def __init__(self, uri, branch, pool, username, password):
+from typing import Optional, Dict, Collection
+from pipeline_dsl.resources.resource import AbstractResource, ConcourseResource
+
+
+class Pool(AbstractResource):
+    def __init__(self, uri: str, branch: str, pool: str, username: str, password: str):
         self.uri = uri
         self.branch = branch
         self.pool = pool
         self.username = username
         self.password = password
 
-    def resource_type(self):
+    def resource_type(self) -> Optional[Dict]:
```

Comment:
Optional - when is this None?

Reply:
This is due to some resources that just return`None` but maybe we should change that to an empty directory in the future.

Reply:
The abstract base class `AbstractResource` defines this method as returning `Optional[Dict]`. To comply with this, the return type is chosen like that. I don't know if it is allowed to drop this "modifier".

> maybe we should change that to an empty directory in the future.

I don't think we should. Some resources return `None` here, because they use the built-in resource types and I think returning `None` in that case is better than returning an empty dict (because it means nothing will be added to the [`resource_types` array](https://concourse-ci.org/pipelines.html#schema.pipeline.resource_types) and not an empty object).